src/pvi/_format/dls.py

```
@@ -2,35 +2,81 @@
 from pathlib import Path
 from typing import List
 
-from typing_extensions import Annotated
+from lxml import etree
 
-from pvi._schema_utils import desc
 from pvi.device import Device
 
 from .base import Formatter
 from .utils import (
     ActionFactory,
+    BobTemplate,
     Bounds,
     EdlTemplate,
     GroupFactory,
     GroupType,
     LabelFactory,
+    LayoutProperties,
     PVWidgetFactory,
     Screen,
+    ScreenWidgets,
     WidgetFactory,
     with_title,
 )
 
 
 @dataclass
 class DLSFormatter(Formatter):
-    spacing: Annotated[int, desc("Spacing between widgets")] = 5
-    title_height: Annotated[int, desc("Height of screen title bar")] = 25
-    max_height: Annotated[int, desc("Max height of the screen")] = 900
-
     def format(self, device: Device, prefix: str, path: Path):
+        if path.suffix == ".edl":
+            f = self.format_edl
+        elif path.suffix == ".bob":
+            f = self.format_bob
+        f(device, prefix, path)
```

You should probably raise an error if you don't support the provided suffix. This is how it used to do so:
`assert path.suffix == ".edl", "Can only write EDL files"`

---

I can see that assertion in format_edl, which is redundant due to the above if / else block. You'll need to rework this bit.

---

see https://github.com/epics-containers/pvi/pull/13/commits/ffa77180c99ca6583005334e049708250bb15cbc